tools/clang/lib/CodeGen/CGHLSLMS.cpp

```diff
@@ -1009,7 +1009,13 @@ unsigned CGMSHLSLRuntime::ConstructStructAnnotation(DxilStructAnnotation *annota
         DxilFieldAnnotation bitfieldAnnotation;
 
         bitfieldAnnotation.SetBitFieldWidth(Field->getBitWidthValue(Context));
-        const BuiltinType *BTy = Field->getType()->getAs<BuiltinType>();
+        QualType FieldTy = Field->getType();
+        const BuiltinType *BTy = FieldTy->getAs<BuiltinType>();
```

Comment:
Other than a `BuiltinType` or an `EnumType` are there other types that could slip in here? Is it possible to get an `AttributedType` here?